cli/cmd/check.go

```diff
@@ -46,6 +48,7 @@ func (options *checkOptions) nonConfigFlagSet() *pflag.FlagSet {
 	flags.StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace to use for --proxy checks (default: all namespaces)")
 	flags.BoolVar(&options.preInstallOnly, "pre", options.preInstallOnly, "Only run pre-installation checks, to determine if the control plane can be installed")
 	flags.BoolVar(&options.dataPlaneOnly, "proxy", options.dataPlaneOnly, "Only run data-plane checks, to determine if the data plane is healthy")
+	flags.BoolVar(&options.dryRun, "dry-run", options.dryRun, "Include checks that rely on k8s dry-run (they may require special write permissions)")
```

Comment:
@grampelberg if we bump our minimum supported K8s version, do you think we should still include this escape hatch to disable dry run checks?  If so, this flag should also apply to the dry run check added in https://github.com/linkerd/linkerd2/pull/3701

Reply:
WDYT about removing the escape hatch for now? If a bunch of folks run into issues with it, it is easy to put back in and in the meantime we don't have to maintain the code unless it is absolutely required.

Reply:
works for me

Reply:
👍 Will remove it then, thanks.